app/src/lib/git/reflog.ts

```diff
@@ -63,3 +63,41 @@ export async function getRecentBranches(
 
   return [...names]
 }
+
+/**
+ * Gets the distinct list of branches that have been checked out after a specific date
+ */
+export async function getCheckoutsAfterDate(
+  repository: Repository,
+  afterDate: Date
+): Promise<ReadonlyArray<string>> {
+  //regexr.com/46n1v
+  const regex = new RegExp(
+    /^(?:.{9})\s(?:\(.*\))?\s?(?:HEAD@){(.*)}:\s(?:.*)to\s(.*)$/
```

Comment:
You might want to look at using `--pretty` to control the output from Git for each reflog entry to make this regex easier to understand. We earlier talked about `--decorate` and how it affects the reflog output:

```
$ git reflog show --date=iso --after="2019-01-17"  --grep-reflog="checkout: moving from .* to .*$" -n 5
4d5591cf3 (origin/port-to-new-parser-for-eslint, port-to-new-parser-for-eslint) HEAD@{2019-01-21 11:55:11 -0400}: checkout: moving from port-to-new-parser-for-eslint to port-to-new-eslint-plugin-for-typescript
4d5591cf3 (origin/port-to-new-parser-for-eslint, port-to-new-parser-for-eslint) HEAD@{2019-01-21 11:48:31 -0400}: checkout: moving from shiftkey/refactor/extract-zoom-logic to port-to-new-parser-for-eslint
a229d899a (origin/development, origin/HEAD, development) HEAD@{2019-01-21 11:38:53 -0400}: checkout: moving from development to tidy-up-zoom-logic
a229d899a (origin/development, origin/HEAD, development) HEAD@{2019-01-21 11:38:47 -0400}: checkout: moving from port-to-new-parser-for-eslint to development
4d5591cf3 (origin/port-to-new-parser-for-eslint, port-to-new-parser-for-eslint) HEAD@{2019-01-21 10:59:54 -0400}: checkout: moving from development to port-to-new-parser-for-eslint
```

Here's an example of how you can avoid the `--decorate` output being rendered:

```
$ git reflog show --date=iso --after="2019-01-17"  --grep-reflog="checkout: moving from .* to .*$" -n 5 --pretty="%h %gd %gs"
4d5591cf3 HEAD@{2019-01-21 11:55:11 -0400} checkout: moving from port-to-new-parser-for-eslint to port-to-new-eslint-plugin-for-typescript
4d5591cf3 HEAD@{2019-01-21 11:48:31 -0400} checkout: moving from shiftkey/refactor/extract-zoom-logic to port-to-new-parser-for-eslint
a229d899a HEAD@{2019-01-21 11:38:53 -0400} checkout: moving from development to tidy-up-zoom-logic
a229d899a HEAD@{2019-01-21 11:38:47 -0400} checkout: moving from port-to-new-parser-for-eslint to development
4d5591cf3 HEAD@{2019-01-21 10:59:54 -0400} checkout: moving from development to port-to-new-parser-for-eslint
```

The placeholders available that are reflog-specific are buried in [this page](https://git-scm.com/docs/git-log#_pretty_formats), if you're looking for something specific:

<img width="647"  src="https://user-images.githubusercontent.com/359239/51487785-4efe1200-1d7a-11e9-8a18-97396f99523c.png">